### PR TITLE
segments example

### PR DESCRIPTION
**Pull request by FelipeAdachi**
## Description

Segments example.
1. segment on single column
2. segment on multiple column
3. segment with filtering
4. writing segments to disk
5. writing segments to WhyLabs

## Changes

- Imperative commit or change title (commit ID) 
  - Add some bullet points to explain the change
- Next commit or change (commit ID)

## Related

Relates to organization/repo#number

Closes organization/repo#number

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/examples/advanced/Segments.ipynb [hunk 1]
```diff
@@ -14,89 +14,1654 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great.\n",
+    "\n",
+    "Now, let's visualize the metrics for a single segment (the first one).\n",
+    "\n",
+    "Results can have multiple partitions, and each partition can have multiple segments. Segments within a partition are non-overlapping. Segments across partitions, however, might overlap. \n",
+    "\n",
+    "In this example, we have only one partition with 11 non-overlapping segments. Let's fatch the available segments:"
```
Comment (jamie256) [inline python/examples/advanced/Segments.ipynb]: typo: fatch -> fetch

File: python/examples/advanced/Segments.ipynb [hunk 2]
```diff
@@ -14,89 +14,1654 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "partition = results.partitions[0]\n",
+    "segments = results.segments_in_partition(partition)"
```
Comment (jamie256) [inline python/examples/advanced/Segments.ipynb]: you should be able to do:
`segments = results.segments()`

And you don't have to bother with the partitions.

**Review by jamie256 — CHANGES_REQUESTED** [review]
I think we need to skip testing this notebook from the automated notebook tests since it asks for user input and interacts with Whylabs, then the CI should pass:

e.g. https://github.com/whylabs/whylogs/actions/runs/3045839526/jobs/4907899428#step:14:114

Otherwise looks great!

File: python/examples/advanced/Segments.ipynb (hunk 1 above)
Comment (FelipeAdachi) [reply python/examples/advanced/Segments.ipynb]: fixed

File: python/examples/advanced/Segments.ipynb (hunk 2 above)
Comment (FelipeAdachi) [reply python/examples/advanced/Segments.ipynb]: updated in latest commit

File: python/test_notebooks/notebook_tests.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "Writing_Reference_Profiles_to_WhyLabs.ipynb",
     "flask_with_whylogs.ipynb",
     "BigQuery_Example.ipynb",
+    "Segments.ipynb",
```
Comment (TheMellyBee) [inline python/test_notebooks/notebook_tests.py:16]: I agree with @jamie256  that I don't think we should test this due to the user input.

**Review by TheMellyBee — DISMISSED** [review]
Looks good! Same change asked as Jamie and it to go green.

**Review by jamie256 — APPROVED** [review]
LGTM! 

Not sure about that last CI failure, looks unrelated to your changes, I'll help look into that, but fine to merge after CI is green.

**Comment by jghoman** [comment]
Super minor question and may be nothing.  For the segments in the screenshot, e.g. "category=Beauty and Hygiene." 

This refers to a single category named "Beauty and Hygiene" and not a segment of those two values (i.e. `select * from <blah> where category = 'beauty' or 'hygiene'`)?  This is what I what I would expect, but it might be a confusing for readers at some point.


Screen Shot 2022-09-13 at 1 44 49 PM

**Pull request merged**